cmake/modules/configuration_files.cmake

```diff
@@ -108,6 +118,7 @@ elseif(EXISTS          ${APPLICATION_CONFIG_DIR}/${BOARD}.overlay)
 elseif(EXISTS          ${APPLICATION_CONFIG_DIR}/app.overlay)
   set(DTC_OVERLAY_FILE ${APPLICATION_CONFIG_DIR}/app.overlay)
 endif()
+zephyr_file_suffix(DTC_OVERLAY_FILE)
```

Comment:
Two issues with this:

1. Suffix may be applied twice, i.e., it's possible to find `boards/board_suffix_suffix.overlay`.
2. If I have an `app_suffix.overlay`, it can't be found unless `app.overlay` also exists.